### ISSUE TITLE
Create [OnParse] attribute to allow run methods as soon as the class is fully-parsed

### ISSUE THREAD

**Issue by fabriciomurta** [bridgedotnet/Bridge#373]
The current Bridge.NET attributes to make classes be automatically called as the script is loaded are limited to events after the page is completely rendered. For instance, the `[Ready]` attribute.

This works well by avoiding class references not to be done until all Bridge.NET scripts have been loaded.

There are some situations though, where this is not true. If the user needs to run a code right after the script is run, as to manipulate some dependency during the page configuration, the only way would be to add a `script` tag right after the tag that loaded the bridge javascript. On this added tag, then insert the full call to the desired method.

For example:
```html
<script src=ClassLibrary1.js></script>
<script>
ClassLibrary1.Class1.myImmediateMethod();
</script>
```

For that matters, it would be interesting to build another attribute (say) `OnParse` to do so without the need to add more code to the referencing page.

One good approach would be like:
C#
```csharp
namespace ClassLibrary1
{
 public class Class1
 {
  [OnParse]
  public static void Main()
  {
   Console.Log("Hello World");
  }
 }
}
```

result in something similar to, in javascript:
```javascript
Bridge.Define('ClassLibrary1.Class1', {
 statics: {
  main: function() {
   console.log("Hello World");
  }
 }
});
ClassLibrary1.Class1.main();
```

In other words, render the same class, but call the methods with `OnParse` just after the class definition.

This would be specially useful for setting up AngularJS controllers during page load.

**Comment by fabriciomurta** [comment]
@vladsch implemented this feature on commits:
@66b07e1c
bridgedotnet/Builder@d729a60b

**Comment by geoffreymcgill** [comment]
I'm not sold on the `[OnParse]` name. I will think about this.

**Comment by erik-kallen** [comment]
I still don't understand why static constructors are not good enough for this feature.

**Comment by vladsch** [comment]
In Bridge v1, static constructor (as any Bridge logic) is invoked after all resources are loaded (we use setTimeout(0) to wait resources loading). The reason: in static constructor can be used a code which is not loaded (not parsed yet).
In Bridge v2 it is not actual, i guess. So, OnParse is for v1 only.

**Comment by geoffreymcgill** [comment]
@vladsch and @fabriciomurta, these revisions need to pulled out of the Master branch, then added to a dedicated feature branch. At that point you can create a pull-request for the feature to be reviewed.

**Comment by fabriciomurta** [comment]
Sorry, thought this particular change wouldn't hit any other feature and might be taken as complete. Is it okay, now that it is already in the code, for we to keep it?

This feature was also briefly discussed during the last meeting and no impediments were raised by then.

**Comment by geoffreymcgill** [comment]
I would prefer these changes be removed from the master-branch and moved to a feature-branch. The current state does not represent the final state, it will have to be revised. At a minimum, the `OnParseAttribute` name will change.

**Comment by fabriciomurta** [comment]
Related branches:
[ ] - bridgedotnet/Bridge@issue373
[ ] - bridgedotnet/Builder@issueBridge373

**Comment by geoffreymcgill** [comment]
Please rename to `[AfterDefine]`.

**Comment by DaniilVeriga** [comment]
Here is a related `[BeforeDefine]` feature issue - #434.

**Comment by geoffreymcgill** [comment]
We've changed the name again to `[Entry]` and the related `[Exit]` Attribute.

**Comment by fabriciomurta** [comment]
And then changed to `[Init]` (still #434).

**Comment by LeonidVeriga** [comment]
Since 434 has been closed we can close this one as well?

**Comment by DaniilVeriga** [comment]
I believe we can. The requested functionality is implemented with a `[Init]` attribute which is a shorthand for `[InitPosition.After]`.
